static/js/publisher/release/actions/releases.js

```diff
@@ -1,5 +1,167 @@
+import {
+  RISKS_WITH_AVAILABLE as RISKS,
+  DEFAULT_ERROR_MESSAGE as ERROR_MESSAGE
+} from "../constants";
+
+import { hideNotification, showNotification } from "./globalNotification";
+import { cancelPendingReleases } from "./pendingReleases";
+import { releaseRevisionSuccess, closeChannelSuccess } from "./channelMap";
+import { updateRevisions } from "./revisions";
+
+import {
+  fetchReleasesHistory,
+  fetchReleases,
+  fetchCloses
+} from "../api/releases";
+
+import { getRevisionsMap, initReleasesData } from "../releasesState";
+
 export const UPDATE_RELEASES = "UPDATE_RELEASES";
 
+function updateReleasesData(releasesData) {
+  return dispatch => {
+    // init channel data in revisions list
+    const revisionsMap = getRevisionsMap(releasesData.revisions);
+    initReleasesData(revisionsMap, releasesData.releases);
+    dispatch(updateRevisions(revisionsMap));
+    dispatch(updateReleases(releasesData.releases));
+  };
+}
+
+export function handleCloseResponse(dispatch, json, channels) {
+  if (json.success) {
+    if (json.closed_channels && json.closed_channels.length > 0) {
+      json.closed_channels.forEach(channel => {
+        // make sure channels without track name get prefixed with 'latest'
+        if (RISKS.indexOf(channel.split("/")[0]) !== -1) {
+          // TODO: This should be the default track, not latest
+          channel = `latest/${channel}`;
+        }
+
+        dispatch(closeChannelSuccess(channel));
+      });
+    }
+  } else {
+    let error = new Error(
+      `Error while closing channels: ${channels.join(", ")}.`
```

Comment:
Hmm.. the only reason to pass `channels` to this function is this error message?
Not a fault of this PR, but it is a bit weird.